Find a Housing Counselor Improvements: Add iteration of results

mistergone opened the pull request:
Update the work-in-progress improvements to Find a Housing Counselor by adding iteration that creates the list of the 10 closest housing counselors.

## Additions
- Add iteration through results, which creates HTML structure for the results of the search

## Testing
1. This only works if the `HUD_TOOL_IMPROVEMENTS` flag is on.
2. Check results at, for example, http://localhost:8000/find-a-housing-counselor/?zipcode=52246
3. Also check the search box

## Screenshots
<img width="1177" alt="screen shot 2018-11-16 at 4 30 44 pm" src="https://user-images.githubusercontent.com/1490703/48648347-01051080-e9bd-11e8-9211-8b03aa3de292.png">

## Checklist
- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -14,6 +14,11 @@
 {% block hero %}
+<script type="text/javascript">
+    var string = "{{ api_json}}";
+    console.log( string );
```
🚒

anselmbradford replied:
Looks like this whole block doesn't need to be there.

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -59,7 +64,7 @@ <h1>Find a housing counselor</h1>
                                         <div class="m-form-field-with-button">
                                             <div class="m-form-field">
-                                                <label class="a-label a-label__heading" for="zip">
+                                                <label class="a-label a-label__heading" for="zipcode">
```
🙇

anselmbradford commented on cfgov/housing_counselor/templates/housing_counselor/index.html:
```diff
@@ -178,7 +178,7 @@ <h1>Find a housing counselor</h1>
                                                     <div class="hud_hca_api_counselor_info_container">
                                                         <div class="hud_hca_api_counselor">
-                                                            {% if valid_url %}<a href="{{ signed_url }}" class="icon-link icon-link__external-link">{% endif %}
+                                                            {% if valid_url %}<a href="????" class="icon-link icon-link__external-link">{% endif %}
```
Set this to `#` and add `TODO` in a comment above it so we can find it in a search.

schbetsy replied:
Seems like we can leave the existing page working as it is, right?

schbetsy commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -14,6 +14,11 @@
+    console.log( string );
+</script>
+
 <section class="m-hero">
```
Would now be a good time to remove the `hero` section, since it is never shown anyway?

anselmbradford replied:
We'll do it in https://github.com/cfpb/cfgov-refresh/pull/4638

mistergone commented:
Apologies - I pushed the wrong branch (I rebased and pushed a fiddling branch instead of the proper one) ! I've fixed that and re-pushed! That should undo those errors.

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -123,41 +125,46 @@ <h2 class="h4">Displaying the 10 locations closest to ZIP code 52246</h2>
+                                            {% endif %}<br>
+                                            {{ counselor.city }}, {{ counselor.statecd }} {{ counselor.zipcd }}
+                                            </p>
+                                            <p><b>Website:</b> <a class="a-link a-link__icon" href="{{ counselor.weburl }}"><span class="a-link_text">{{ counselor.weburl }}</span> <svg class="cf-icon-svg" viewbox="0 0 790.1 1200" xmlns="http://www.w3.org/2000/svg">
```
Let's break this out into multiple lines.

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -123,41 +125,46 @@ <h2 class="h4">Displaying the 10 locations closest to ZIP code 52246</h2>
-                                            <path d="M740.1 178.8H449.5c-27.6 0-50 22.4-50 50s22.4 50 50 50h169.9L518.2 379.9c-14.7-6.2-30.5-9.4-46.4-9.4H120c-66.2.1-119.9 53.8-120 120v351.8c.1 66.2 53.8 119.9 120 120h351.8c66.2-.1 119.9-53.8 120-120V490.6c0-12.6-2-25-5.8-37l104.1-104.1v169.9c0 27.6 22.4 50 50 50s50-22.4 50-50V228.8c0-27.7-22.4-50-50-50zM491.8 842.3c-.1 11-9 19.9-20 20H120c-11-.1 ...
-                                            Iowa City, IA 52240-8132</p>
-                                            <p><b>Website:</b> <a class="a-link a-link__icon" href=""><span class="a-link_text">http://www.example.com</span> <svg class="cf-icon-svg" viewbox="0 0 790.1 1200" xmlns="http://www.w3.org/2000/svg">
+                                            <path d="M740.1 178.8H449.5c-27.6 0-50 22.4-50 50s22.4 50 50 50h169.9L518.2 379.9c-14.7-6.2-30.5-9.4-46.4-9.4H120c-66.2.1-119.9 53.8-120 120v351.8c.1 66.2 53.8 119.9 120 120h351.8c66.2-.1 119.9-53.8 120-120V490.6c0-12.6-2-25-5.8-37l104.1-104.1v169.9c0 27.6 22.4 50 50 50s50-22.4 50-50V228.8c0-27.7-22.4-50-50-50zM491.8 842.3c-.1 11-9 19.9-20 20H120c-11-.1 ...
```
Indentation is wonky here. `<svg>` is indented more than `<path>`.

anselmbradford replied:
Hold on... why is `svg` here at all? This should be `{{ svg_icon('external-link') }}`

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -123,41 +125,46 @@ <h2 class="h4">Displaying the 10 locations closest to ZIP code 52246</h2>
-                                            <b>Languages:</b> English</p>
+                                            <b>Phone:</b> {{ counselor.phone1 }}<br>
+                                            <b>Email Address:</b> {{ counselor.email }}<br>
+                                            <b>Languages:</b> {{ counselor.languages|join(', ') }}</p>
```
TRIVIAL: We don't stick to this, but in general I think the pipe and parens are spaced: `{{ counselor.languages | join( ', ' ) }}`

anselmbradford commented on cfgov/housing_counselor/jinja2/housing_counselor/improved_index.html:
```diff
@@ -123,41 +125,46 @@ <h2 class="h4">Displaying the 10 locations closest to ZIP code 52246</h2>
                                     <td data-label="Distance">
                                         <div class="rich-text">
-                                            <p>3.1 miles</p>
+                                            <p>{{ counselor.distance|round( 1 ) }} miles</p>
```
TRIVIAL: space the pipe if you space the parens: `{{ counselor.distance | round( 1 ) }}`

anselmbradford commented:
Minor formatting issues that could use cleanup, but otherwise LGTM.
